src/cargo/util/toml/mod.rs

```diff
@@ -1595,7 +1595,42 @@ pub struct InheritableFields {
     ws_root: PathBuf,
 }
 
+/// Defines simple getter methods for inheritable fields.
+macro_rules! inheritable_field_getter {
+    ( $(($path:literal, $field:ident -> $ret:ty),)* ) => (
+        $(
+            #[doc = concat!("Gets the field `workspace.", $path, "`.")]
+            pub fn $field(&self) -> CargoResult<$ret> {
+                if let Some(val) = &self.$field {
+                    return Ok(val.clone())
+                }
+                bail!("`workspace.{}` was not defined", $path);
```

Comment:
Nit: A `let-else` might be better as it would eliminate the early `return`. It would also move `bail!` into an `else`, which feels more explicit than falling through when `None` is encountered.
```suggestion
               let Some(val) = &self.$field else  {
                    bail!("`workspace.{}` was not defined", $path);
                };
                Ok(val.clone())
```